lib/active_model/serializer/adapter/json.rb

```diff
@@ -47,6 +25,61 @@ def fragment_cache(cached_hash, non_cached_hash)
           Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
         end
 
+        private
+
+        # iterate through the associations on the serializer,
+        # adding them to @hash as needed (as singular or plural)
+        def add_resource_relationships(serializer)
+          serializer.associations.each do |association|
+            serializer = association.serializer
+            opts = association.options
+
+            if serializer.respond_to?(:each)
+              add_relationships(association.key, serializer, opts)
```

Comment:
I know the terminology comes from the `json_api` adapter, but I think `add_to_many_relationship` and `add_to_one_relationship` are more intuitive name than the confusingly similar `add_relationship`/`add_relationships`.